src/libs/blueprint/conduit_blueprint_mesh_partition.cpp

```diff
@@ -8635,33 +8638,104 @@ determine_schema(const Node &in,
     if(num_children)
     {
         out_ncomps = num_children;
-        index_t offset = 0;
-        // TODO: Keep track of whether the original field was interleaved
-        //  and preserve the interleaved-ness
+
+#if defined(CONDUIT_DEBUG_PARTITIONER_MAPBACK) && defined(CONDUIT_DEBUG_PARTITIONER_MAPBACK_VERBOSE)
+        std::cout << "========================================================" << std::endl;
+        std::cout << "determine_schema:" << std::endl;
         for(index_t i = 0; i < num_children; i++)
         {
-            const DataType dt(in[i].dtype().id(), ntuples, offset,
-                in[i].dtype().element_bytes(), in[i].dtype().element_bytes(),
-                in[i].dtype().endianness());
-            out_schema[in[i].name()].set(dt);
-            offset += dt.number_of_elements() * dt.element_bytes();
+            std::cout << "  -" << std::endl;
+            std::cout << "    name: " << in[i].name() << std::endl;
+            in[i].dtype().to_string_stream(std::cout, "yaml", 4, 1);
+        }
+        std::cout << "========================================================" << std::endl;
+#endif
+
+        // Check that all children are the same type.
+        bool same_types = true;
+        for(index_t i = 1; i < num_children; i++)
+            same_types = (in[i].dtype().id() == in[0].dtype().id());
+
+        // Check whether components appear interleaved.
+        bool interleaved = true;
+        for(index_t i = 1; i < num_children; i++)
+        {
+            const auto dI = in[i].dtype().element_index(0) - in[i - 1].dtype().element_index(0);
+            interleaved &= (dI == in[i].dtype().element_bytes());
+        }
```

Comment:
can mcarray is_interleaved helper work here?

https://github.com/LLNL/conduit/blob/develop/src/libs/blueprint/conduit_blueprint_mcarray.cpp#L252

Reply:
Yes. Thanks. I made the change.